### PR TITLE
Add proxy support to the dataset container #2512

### LINKED ISSUE #2512: Enable proxy settings for dataset in values of helm
**Is your feature request related to a problem? Please describe.**
The dataset container of graphscope provides the function of obtaining sample dataset online, but there are many inconveniences in accessing the external network to obtain data in many Intranet environments. Therefore, the option of configuring the proxy service address for the dataset container should be added to the configuration file of helm

**Describe the solution you'd like**
In the values file of helm, add the configuration item proxy of dataset.

**Describe alternatives you've considered**
I provided the corresponding configuration of dataset in the values file of helm, modified the cluster_builder.py coordinator.py kubernetes_launcher.py, and added a dataset_proxy attribute in the corresponding classes

**Additional context**

### PR DESCRIPTION
**Pull request by YANGBoSunning**
This PR adds an option of configuring the proxy service address for the dataset container should be added to the configuration file of helm

The dataset container of graphscope provides the function of obtaining sample dataset online, but there are many inconveniences in accessing the external network to obtain data in many Intranet environments. 

Fixes #2512

### CODE REVIEW

File: charts/graphscope/values.yaml [hunk 1]
```diff
@@ -70,6 +70,9 @@ engines:
     enabled: False
     image:
       name: dataset
+#    proxy:
```
Comment (siyuan0322) [inline charts/graphscope/values.yaml:73]: Could you add a comment here for what is `proxy` for, and what's the target scenario, it makes people or developers like me could understand this more easily

File: coordinator/gscoordinator/coordinator.py [hunk 2]
```diff
@@ -880,6 +880,14 @@ def parse_sys_args():
         default=9968,
         help="Coordinator prometheus exporter service port.",
     )
+    parser.add_argument(
+        "--dataset_proxy",
+        type=str,
+        nargs="?",
+        const="{}",
+        default="{}",
+        help="Proxy for worker dataset."
```
Comment (siyuan0322) [inline coordinator/gscoordinator/coordinator.py:889]: And please make this help more meaningful. That would be very helpful.

File: coordinator/gscoordinator/cluster_builder.py [hunk 3]
```diff
@@ -144,6 +145,8 @@ def __init__(
         self._preemptive = preemptive
         self._vineyard_shared_mem = vineyard_shared_mem
 
+        self._dataset_proxy = dataset_proxy
```
Comment (siyuan0322) [inline coordinator/gscoordinator/cluster_builder.py:148]: I prefer to let the `dataset_proxy` follow the logic of `volumes`, they are both json strings. i.e. 
- loads the str to json in here, not in the `EngineCluster(dataset_proxy=json.loads(dataset_proxy), ...)`; 
- b64encode it in the chart then decode here.

**Review by siyuan0322 — APPROVED** [review]

**Comment by YANGBoSunning** [comment]
1. add comment for dataset proxy in values.yaml
2. add help info for argument --dataset_proxy
3. change logic of dataset_proxy, input with base64 and decode in EngineCluster's init function
@siyuan0322

**Comment by siyuan0322** [comment]
LGTM, we could merge this after the CI check.

**Comment by siyuan0322** [comment]
Merged. Thanks for your contribution @YANGBoSunning !

**Comment by YANGBoSunning** [comment]
Thanks for your review, good luck!

**Comment by YANGBoSunning** [comment]
I'm sorry about that i missed one line code in last commit and i have add it in branch YANGBoSunning:pr-2512. Please fix it. @siyuan0322

**Comment by siyuan0322** [comment]
> I'm sorry about that i missed one line code in last commit and i have add it in branch YANGBoSunning:pr-2512. Please fix it. @siyuan0322

Got it 😂 , thank you

**Pull request merged**